### ISSUE TITLE
SyntaxError: Cannot convert  exponential number to a BigInt

### ISSUE THREAD

**Issue by Trudko** [beenotung/compress-json#9]
Compress-JSON is converting number to a [Big Int](https://github.com/beenotung/compress-json/blob/master/src/number.ts#L103). BigInt doesn't work with exponential numbers in string form so

```
import { compress } from "compress-json";

const data = {
    a: 1.23456789123789e22,
};

compress(data);
```
will fail with:
```
SyntaxError: Cannot convert 22+e98732198765432 to a BigInt
    at BigInt (<anonymous>)
    at int_str_to_s (/Users/my-user/test/node_modules/compress-json/dist/number.js:105:31)
    at Object.num_to_s (/Users/my-user/test/node_modules/compress-json/dist/number.js:96:9)
    at Object.encodeNum (/Users/my-user/test/node_modules/compress-json/dist/encode.js:6:31)
    at addValue (/Users/my-user/test/node_modules/compress-json/dist/memory.js:145:46)
    at Object.addValue (/Users/my-user/test/node_modules/compress-json/dist/memory.js:137:31)
    at compress (/Users/my-user/test/node_modules/compress-json/dist/core.js:9:27)

```

**Comment by beenotung** [comment]
Thanks for the report, already having idea on how to fix it

**Comment by beenotung** [comment]
The reported bug is fixed in https://github.com/beenotung/compress-json/commit/f9c83c99b67ceb45be8f8e5b8f389cda88b42f64 and the patched version is release as compress-json@2.1.2

@Trudko you may update the package version then confirm if it works for your case :rocket:

**Comment by Trudko** [comment]
@beenotung works like a charm :)
